### DISCUSSION TITLE
Self.I problem?

### DISCUSSION THREAD

**Discussion by hz7451** [kernc/backtesting.py#545] [Q&A] [answered]
I have a RSI indicator inside `def init(self)` and print out result like this `
[nan 100  48  20  18  16  13  12  46  54]`. If i do not add RSI indicator, the `def next()` will perform "normally" which will start from the third candle and so on. Once I added RSI, the `def next()` will start from forth candle maybe due to the first RSI data "nan" ? I am not sure . Can I add RSI indicator at the same time the `def next()` works as normal and start from the third candle?

**Comment by kernc** [comment, marked as answer]
> maybe due to the first RSI data "nan" ?

Exactly. Just fill the nan with some value, e.g.:
```py
def RSI(*args):
    rsi = _RSI(*args)
    return pd.Series(rsi).fillna(method='backfill')
```
